### PR TITLE
docker helper improve

### PR DESCRIPTION
**Pull request by esharf**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes some container build exceptions.

## Description

- code design looks better :sunglasses:.
- we can now build docker images in the circleCI environment.

circle-build: https://app.circleci.com/pipelines/github/demisto/content/152793/workflows/49ed8318-ae3f-42fe-93c5-ffa950b13cae/jobs/498050
gitlab-build: https://code.pan.run/xsoar/content/-/jobs/11719406

## Must have
- [ ] Tests

### CODE REVIEW

**Review by yaakovi — COMMENTED** [review]
@wissamg / @jochman - please have a look too (one of you is enough)

**Review by wissamg — APPROVED** [review]

File: demisto_sdk/commands/lint/docker_helper.py [hunk 1]
```diff
@@ -125,36 +128,61 @@ def create_image(base_image: str, image: str, container_type: str = TYPE_PYTHON,
...
+        kwargs = kwargs or {}
+        if files_to_push and mount_files:
+            return super(MountableDocker, self).create_container(image=image, command=command, environment=environment,
+                                                                 mounts=self.get_mounts(files_to_push), files_to_push=None, **kwargs)
+        else:
+            return super(MountableDocker, self).create_container(image=image, command=command, environment=environment,
+                                                                 files_to_push=files_to_push, **kwargs)
+
+
+Docker = MountableDocker() if CAN_MOUNT_FILES else DockerBase()
```
[the remark this answers is not in the document]
Comment (esharf) [reply demisto_sdk/commands/lint/docker_helper.py:191]: this is intentional

File: demisto_sdk/commands/lint/docker_helper.py [hunk 2]
```diff
@@ -44,7 +48,29 @@ def copy_file(cp_from: PATH_OR_STR, cp_to: PATH_OR_STR) -> Path:
...
+            TYPE_PWSH: installation_scripts / 'powershell_image.sh',
+        }
+        self.changes = {
+            TYPE_PWSH: ['WORKDIR /devwork'],
+            TYPE_PYTHON: ['WORKDIR /devwork', 'ENTRYPOINT ["/bin/sh", "-c"]'],
+        }
+        self.requirements = self.tmp_dir / 'requirements.txt'
+        self.requirements.touch()
+        self._files_to_push_on_installation: FILES_SRC_TARGET = [
+            ('/test-requirements.txt', self.requirements),
```
[the remark this answers is not in the document]
Comment (esharf) [reply demisto_sdk/commands/lint/docker_helper.py]: intentional

**Comment by dorschw** [comment]
fyi @ilaner @jochman

**Pull request merged**